cosmo_tester/test_suites/snapshots/single_tenant_snapshot_test.py

```diff
@@ -67,7 +68,9 @@ def test_restore_snapshot_and_agents_upgrade_singletenant(
 
     assert_hello_worlds([hello_vm], installed=True, logger=logger)
 
-    check_secrets_converted(new_manager, logger)
+    if old_manager.branch_name in ('3.4.2', '4.0'):
+        # Check we convert secrets for pre-service-user versions
```

Comment:
can you also write "the why" in this comment please? i'm not sure, do we not support converting secrets in singletenant post-4.0?